provenance.jq

```diff
@@ -0,0 +1,76 @@
+# input: "build" object (with "buildId" top level key)
+# output: list of image tags
+def tags:
+	[
+		.source.arches[].tags[],
+		.source.arches[].archTags[],
+		.build.img
+	]
+;
+
+# input: "tags" object with image digest and platform arguments
+# output: json object for in-toto provenance subject field
+def subjects($platform; $digest):
+	($digest | split(":")) as $splitDigest |
+	{
+		"name": ("pkg:docker/" + . + "?platform=" + $platform),
+		"digest": {
+			($splitDigest[0]): $splitDigest[1]
+		}
+	}
+;
+
+# input: GITHUB context argument
+# output: json object for in-toto provenance external parameters field
+def github_external_parameters($github):
+	($github.workflow_ref | gsub( $github.repository + "/"; "")) as $workflowPathRef |
+	{
```

Comment:
`gsub` does a regex match, so I think we should probably use `ltrimstr` instead to trim the prefix instead, and probably `split` up here too (so we don't do that twice) -- it's also wild, yet again, that `workflow_ref` is the best granularity they offer here and it's this weird string that's unlike anything used anywhere else in the GitHub platform (`octocat/hello-world/.github/workflows/my-workflow.yml@refs/heads/my_branch`) :sob:

```suggestion
	($github.workflow_ref | ltrimstr($github.repository + "/") | split("@")) as $workflowRefSplit
	| {
```